### PR TITLE
set strict window for couch compaction on ICDS

### PR DESCRIPTION
**Pull request by snopoke**
##### SUMMARY
Prevent CouchDB compaction from continuing beyond allocated time period.

##### ENVIRONMENTS AFFECTED
ICDS

##### ISSUE TYPE
- Change Pull Request

##### COMPONENT NAME
CouchDB

##### ADDITIONAL INFORMATION
https://docs.couchdb.org/en/stable/config/compaction.html#compaction-daemon-rules

### CODE REVIEW

File: environments/icds-cas/public.yml [hunk 1]
```diff
@@ -145,6 +145,9 @@ couchdb_cluster_settings:
   w: 2
   n: 2
 
+couchdb_compaction_settings:
+  _default: '[{db_fragmentation, "10%"}, {view_fragmentation, "10%"}, {from, "17:00"}, {to, "05:00"}, {strict_window, true}]'
```
Comment (mkangia) [inline environments/icds-cas/public.yml]: is 17:00 and 5:00 will respect UTC or IST?

Comment (snopoke) [reply environments/icds-cas/public.yml]: Machine time, so it should be UTC (22:00 to 10:00 IST).

**Comment by snopoke** [comment]
@emord good question - currently our compaction mostly complete at about 05:30 which is 12.5 hours after starting. Ideally I'd like to have them end a bit earlier so that we're well out of peak but I'm not sure what the consequences of that will be in the long term. I was thinking of deploying this and seeing how it goes.

**Comment by emord** [comment]
That seems fine. Though long term we need to get these on ubuntu 18.04 and couch 2.3.1 to fix the issue where its only compacting the first shard. Was found by @dannyroberts a while ago, ended up saving us a lot of money hosting couch for production.

**Comment by snopoke** [comment]
Rollout ticket: https://dimagi-dev.atlassian.net/browse/IIO-369

**Pull request merged**